### ISSUE TITLE
Request - please help test a form - authority picker

### ISSUE THREAD

**Issue by dustymc** [ArctosDB/arctos#9366]
Help us understand your request (check below):

- [x] other

**Describe what you're trying to do** 

I'd like some early feedback on https://github.com/ArctosDB/dev/issues/460. There is a very early and very limited instance of this beginning to become functional in test, I could use a sanity check.

It works only on new attributes under edit locality in test at this time.

Get started in the usual way by selecting a type

<img width="214" height="299" alt="Image" src="https://github.com/user-attachments/assets/d1ee525d-bc40-4515-81f6-38b208a32fde" />

SIDE NOTE:   I've also added a way to open the code table from the chooser, select...

<img width="184" height="54" alt="Image" src="https://github.com/user-attachments/assets/46e9ab4d-bbf4-4246-8ce4-cb78cf474a5f" />

and you'll get a full-viewport overlap of the "base" authority...

<img width="1690" height="325" alt="Image" src="https://github.com/user-attachments/assets/742a3d2a-e818-47d3-a33f-7476f37391d3" />

Free-text (or value+unit) attributes types haven't changed:

<img width="381" height="77" alt="Image" src="https://github.com/user-attachments/assets/fe547666-eb23-4d6f-8e80-9dce182b079d" />

If a categorical attribute type is chosen, the form will change to...

<img width="680" height="74" alt="Image" src="https://github.com/user-attachments/assets/84e8c491-1ff7-4ada-97a6-4f2bccdf9e8d" />

with the value focused - just start typing if you (kinda) know what you want, then TAB to open the pick, or click "pick" to open the chooser without attempting to filter.

Either way, you'll get a popup with three sections:

first node documentation

<img width="405" height="148" alt="Image" src="https://github.com/user-attachments/assets/9a730929-b63f-4489-9683-0dceaa58ce7a" />

then a skinny scrolley window of type documentation (so it's easy to confirm that you've picked the right attribute type)

<img width="1683" height="261" alt="Image" src="https://github.com/user-attachments/assets/600dceb6-92f1-4725-bef8-b8b5a27765e6" />

and finally values (with, in this case, terrible documentation..)

<img width="540" height="320" alt="Image" src="https://github.com/user-attachments/assets/fd23332b-7d68-4a94-ab5a-ad807ffd99ef" />

If the form was opened by type-and-tab, that will be in the search box, and matches will be highlighted. You can also type in the search box then hit TAB or click the button to re-query, or just scroll around and pick something. 

All values will always be somewhere in your window where any search tool can find them, which will hopefully put everything in context of everything else without being too overwhelming.

The pick-window should scroll to the first search result, if there are any - and here's a "busier" view:

<img width="1706" height="671" alt="Image" src="https://github.com/user-attachments/assets/daa006f8-6db6-4ee4-89be-892f2e662a5a" />

Find something you like, click the button, the calling form value is filled in, the overlay is closed, and the cursor is focused in the next pick (determiner).

I think this will (usually) be fewer clicks than the old pick is at the moment.

I think I've included all the relevant information needed to fully understand the data.

I think the form is "sprinklable" and can be used to pick other stuff (and parts should probably be the next adaptation).

My browser is happy enough with 3177 quads, although there is a bit of a pause as everything is processed. I think this is as scalable as it can be without paging, and paging would make the data less accessible.

There's a LOT of information on the pick-page, but I think there needs to be, and I suspect most users will quickly learn where things are when they need them, and how to ignore that when they don't. The code is pretty generic, a UI/UX expert should be able to get a long ways with CSS and it should be easier to cut deeper if necessary; I'm pretty confident that the base is solid.

Please let me know ASAP if anything seems wildly wrong.

------------

Please review the [Arctos Code of Conduct](https://arctosdb.org/code-of-conduct/) to ensure respectful communication, and see <https://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html> for more information about submitting and managing Issues.

**Comment by Nicole-Ridgwell-NMMNHS** [comment]
I like this. My only slight nitpick is that when I type in a value and go to pick, the bar at the top covers up the actual match:

<img width="1901" height="162" alt="Image" src="https://github.com/user-attachments/assets/a61c2503-d3f9-4a56-8f80-0d860c0b1dac" />

**Comment by mkoo** [comment]
yes, I agree-- nice improvement now that I've used it a few times! great way to filter/find values. I havent had time to show curatorial assistants but will next week so there may be more input. Thx!

**Comment by Jegelewicz** [comment]
@dustymc I have a request.

This picker provides a lot of info, but still requires a lot of scrolling. So, if I want to pick the lithostratigraphic formation "Nacimiento Formation" and I use the pick button because I don't know how to spell it, I get 5 results in the picker.

<img width="1507" height="769" alt="Image" src="https://github.com/user-attachments/assets/4b251cbc-f2ea-4c29-8b5b-70b5057df398" />

and the 5 things are highlighted, but there are a bazillion rows to scroll through just to find the highlighted things

can the results be condensed to the actual 5 results?

**Comment by dustymc** [comment]
>can the results be condensed to the actual 5 results?

Part of how we got here is the possibility of Nacimiento Formation Nacimiento Sandstone Formation and Nacimiento Whatever Formation and Nacimiento Not-Formation all co-existing, so you search 'Nacimiento Sa' and see one, don't see the other 27, assume you've found what you need because Arctos has removed the context of those things existing, and pick the wrong thing. One possible way to avoid that is to just always include everything, which is where we are now.

I played with putting matches at the top of the list (still removes them from likely context) and some other stuff, but didn't some up with anything more clever than the matchcount-report-thing (which ain't that clever...).

<img width="79" height="41" alt="Image" src="https://github.com/user-attachments/assets/cd055720-511f-48be-aebf-33e536d3e9b3" />

I'm up for whatever, and the "everything everywhere" approach might not survive the reality of thousands more formations anyway. It also doesn't address the flakiness of the rank - some things you might think should be formations may have been created as members (etc), and this won't find those very well - something to keep in mind in case there's some clever idea that covers it.

(And sorry I'm not at the meeting to discuss this right now, I have to deal with some higher-level stuff, will try to multitask issues though.)

**Comment by Jegelewicz** [comment]
> some things you might think should be formations may have been created as members (etc), and this won't find those very well - something to keep in mind in case there's some clever idea that covers it.

Have one attribute: lithostratigraphy

Include in the code table a column with the "types" and allow for multiple values ( for when things get moved from one type to another - it happens and everyone may not know it)

Remove the type from the values 
or if that makes people nuts, leave it and you'll just have two very similar names in the table (and probably pick the wrong one)

Sure there will a ton of things in the code table, but when a member gets bumped up to a formation, the only thing that needs to happen is adding "formation" to the type column in the code table. The types could be used for filtering in pick lists or not.

**Comment by Jegelewicz** [comment]
> Part of how we got here is the possibility of Nacimiento Formation Nacimiento Sandstone Formation and Nacimiento Whatever Formation and Nacimiento Not-Formation all co-existing, so you search 'Nacimiento Sa' and see one, don't see the other 27, 

This is starting to feel like treating collections staff like little kids who can't make proper decisions. At some point, we need to let go of being always right and allow people to make mistakes. They can be corrected.

**Comment by dustymc** [comment]
I think we're done here, comments have all been addressed or involve behavior which can easily be changed wherever the UI is deployed. I'm closing this and incrementing implementation to all of edit locality.

Direction/guidance still greatly appreciated, feel free to use https://github.com/ArctosDB/dev/issues/460 for minor comments/problems, or file a new Issue to discuss major change (eg how search works).

**Comment by Jegelewicz** [comment]
Reopening to post comments from today's AWG which I believe are minor revisions.

1. When displaying all options, for those not available to a collection instead of the use button, offer a simple message such as "Not selected for collection"
2. Add to the documentation on the top of the Authority Picker screen that refers to the managed collection for the code table. "To add terms for use in this collection, go to the code table and select the term you wish to use. If you do not have manage collection permissions, ask your supervisor to add the term."

**Comment by campmlc** [comment]
This is clear. I support this.

**Comment by dustymc** [comment]
>Reopening

I think there's no chance anyone will find a way to turn a comment buried in a test request into a functional need/dev issue. Suggest a new Issue.
